README.md

```diff
@@ -78,225 +78,791 @@ These decisions enable the following properties/ behaviors:
-| `dmx_prefix`       | 10      | prefix for versioning packet formats (default: `tinyssb-v0`) ?? TODO: Encoding? |
-| `feed_id`          | 32      | ed25519 public key                                                              |
-| `sequence`         | 4       | the sequence of this packet in the feeds linked-list, in big-endian format      |
-| `prev_message_id`  | 20      | the id of the previous message in this feeds linked-list                        |
+#### 2.2.3 CLAIM Packet Format
 
+The GOSET protocol uses CLAIM packets to synchronize the set of Feed IDs between peers. A CLAIM packet describes a region of the lexicographically sorted GOSET.
 
+A CLAIM packet has the following structure:
+
+```
+  |<------------------------------ 103 bytes ------------------------------>|
+  
+  ┌──────────┬──────┬────────────────┬────────────────┬────────────────┬────┐
+  │ DMX      │ 'c'  │ Lowest FID     │ Highest FID    │ XOR            │ cnt│
+  │ (7 bytes)│(1 B) │ (32 bytes)     │ (32 bytes)     │ (32 bytes)     │(1B)│
+  └──────────┴──────┴────────────────┴────────────────┴────────────────┴────┘
+```
+
+Where:
+- **DMX**: First 7 bytes of `SHA256(GOSET_DMX_STR)`, where `GOSET_DMX_STR = "tinySSB-0.1 GOset 1"`
+- **Type**: Single byte with value 'c' (0x63) indicating a CLAIM packet
+- **Lowest FID**: The lexicographically lowest Feed ID in the region
+- **Highest FID**: The lexicographically highest Feed ID in the region
+- **XOR**: The XOR-sum of all Feed IDs in the region
+- **cnt**: The number of Feed IDs in the region (1-255)
+
+The total length of a CLAIM packet is 103 bytes, which fits within the 120-byte limit.
+
+### 2.3 GOSET Synchronization Process
+
+The GOSET synchronization process allows peers to identify and share missing Feed IDs. It works through a series of CLAIM packet that progressively narrow down regions where peers have different sets.
+
+#### 2.3.1 Basic Process
+
+1. A peer broadcasts a CLAIM packet covering its entire GOSET
+2. If another peer has a different set of Feed IDs in that region (detected by comparing the XOR and count), it responds with its own CLAIM for the same region
+3. Once a peer identifies a missing Feed ID, it adds it to its GOSET
+4. The first peer then splits the region into two sub-regions and sends CLAIM packets for each
+5. This process continues recursively, narrowing down to the specific Feed IDs that differ
+
+```mermaid
+sequenceDiagram
+    participant A as Peer A
+    participant B as Peer B
+    
+    A->>B: CLAIM (region: all known FIDs)
+    Note over B: B adds any missing to GOSET
+    B->>A: CLAIM (region: all known FID)
+    A->>B: CLAIM (sub-region 1)
+    A->>B: CLAIM (sub-region 2)
+    Note over A,B: Repeat till CLAIMs align...
+```
+
+#### 2.3.2 Detecting Differences
+
+Peers detect differences in their GOSETs by comparing the `XOR`-sum and `count` of Feed IDs in a region:
```

Comment:
```suggestion
Peers detect differences in their GOSETs by comparing the `XOR`-sum and `count` of Feed IDs in a region. Before comparing these values, however, a peer adds the CLAIM's first and last Feed IDs to its own set in case they are not already present.
```